service/auth_service.go

```diff
@@ -106,15 +95,25 @@ func (a *AuthService) RefreshTokens(refreshToken string) (*authProto.GeneratedTo
 	return a.GenerateTokensByAuthUser(&authProto.User{UserId: claims.UserId, Role: role.Name})
 }
 
-func (a *AuthService) CheckRights(token string, requiredRole string) (bool, error) {
-	userRole, err := a.ParseToken(token)
-	if err != nil {
-		return false, err
+func (a *AuthUserService) CheckRoleRights(neededPerms []string, neededRole string, givenPerms string, givenRole string) error {
+	if neededPerms != nil {
+		ok := true
+		for _, perm := range neededPerms {
+			if !strings.Contains(givenPerms, perm) {
+				ok = false
+				return fmt.Errorf("not enough rights")
+			} else {
+				continue
+			}
+		}
+		if ok == true {
```

Comment:
I would honestly just separated checking roles and permissions, because in many cases you don't need checking permissions at all, but only roles. And it would just make the logic of this function more clear even not only in terms of code, but in terms of expected result (when you pass both roles and permissions do you expect both of them match or only one of them?).

Reply:
I did it